courseworkA/Markdown report template assignment A .Rmd

```diff
@@ -55,197 +55,293 @@ Describe which participants will recruit in the study and how they will be recru
+# # by substracitng the number of occurrence of negative words from that of positive words
+# 
+# analysis_T <- score.sentiment(tweets_T.text, pos, neg)
+# analysis_C <- score.sentiment(tweets_C.text, pos, neg)
+# analysis_B <- score.sentiment(tweets_B.text, pos, neg)
+# analysis_A <- score.sentiment(tweets_A.text, pos, neg)
+# analysis_J <- score.sentiment(tweets_J.text, pos, neg)
+# 
+# 
+# sem<-data.frame(analysis_T$score, analysis_C$score, analysis_B$score, analysis_J$score, analysis_A$score)
+# 
+# 
+# semFrame <-melt(sem, measured=c(analysis_T.score,analysis_C.score, analysis_B.score ))
+# names(semFrame) <- c("Celeb", "score")
+# semFrame$Celeb <-factor(semFrame$Celeb, labels=c("Donald Trump", "Tucker Carlson","Bernie Sanders", "Alexandria Ocasio-Cortez ", "Joe Biden")) # change the labels for your celibrities
+
+semFrame <- readRDS(file="semFrame.Rda")
+
+
+#saveRDS(semFrame, file="semFrame.Rda")
+#The data you need for the analyses can be found in semFrame
 
-<!-- tweets_T.text <- laply(tweets_T, function(t)t$getText()) #get text out of tweets  -->
-<!-- tweets_C.text <- laply(tweets_C, function(t)t$getText()) #get text out of tweets -->
-<!-- tweets_B.text <- laply(tweets_B, function(t)t$getText()) #get text out of tweets -->
+```
 
+### Homogeneity of variance analysis
+```{r}
+pander(leveneTest(semFrame$score, semFrame$Celeb))
+```
 
+The Levene test reveals a p-value smaller than 0.001, indicating that there is significant difference between the group variances in sentiment score. We conclude that the variance among the five groups is not equal.
+
+### Visual inspection Mean and distribution sentiments
+We plot both a line density and a distribution histogram which includes a mean line.
+```{r}
+#boxplot(score ~ Celeb, data = semFrame, text.font = 4)
+sm.density.compare(semFrame$score, semFrame$Celeb, xlab = "Score",
+                   col=c('red', 'blue', 'green', 'hotpink', 'gold'), lty=c(1,1,1,1,1))
+  title(main="Visual inspection Mean and distribution sentiments")
+  legend('topright', legend = levels(semFrame$Celeb),
+         col=c('red', 'blue', 'green', 'hotpink', 'gold'),
+         title="Celebs", lty=1, cex=0.8, text.font = 4, bg='lightblue')
+
+cdat <- ddply(semFrame, "Celeb", summarise, score.mean=mean(score))
+ggplot(semFrame, aes(x=score, fill=Celeb)) +
+  geom_histogram(binwidth=1, position="dodge") +
+  geom_vline(data=cdat, aes(xintercept=score.mean,  colour=Celeb),
+               linetype="solid", size=0.5) 
 
-<!-- #taken from https://github.com/mjhea0/twitter-sentiment-analysis -->
-<!-- pos <- scan('positive-words.txt', what = 'character', comment.char=';') #read the positive words -->
-<!-- neg <- scan('negative-words.txt', what = 'character', comment.char=';') #read the negative words -->
+```
+We see all US politic Celebs have a mean around 0. #trump has the highest sentiment mean and the largest difference with the rest.
```

Comment:
In the pdf the "We" is in the line next to the image. Maybe start a new line to avoid that